src/imap/mod.rs

```diff
@@ -1711,6 +1712,66 @@ fn get_fallback_folder(delimiter: &str) -> String {
     format!("INBOX{}DeltaChat", delimiter)
 }
 
+/// Builds a list of sequence/uid sets. The returned sets have each no more than around 1000
+/// characters because according to https://tools.ietf.org/html/rfc2683#section-3.2.1.5
+/// command lines should not be much more than 1000 chars (servers should allow at least 8000 chars)
+fn build_sequence_sets(mut uids: Vec<u32>) -> Vec<String> {
+    uids.sort_unstable();
```

Comment:
Now sort_unstable() is called on every fetch, even though uids are already sorted. However, relying on this seemed very fragile as you have to read to 3 functions to confirm that this always is the case. Also, this makes the `server_uids.clone()` necessary because it is not enough to just borrow `server_uids`.

The slighter faster but IMHO more fragile alternative would be to drop this sort, let `built_sequence_sets()` take a slice again and only sort in `get_all_recipients()`.

Reply:
I guess CPU time required to sort a few numbers is negligible.